src/app/api/auth/signup/route.ts

```diff
@@ -0,0 +1,62 @@
+import { prisma } from "@/lib/prisma";
+import {
+  emailValidationSchema,
+  passwordValidationSchema,
+} from "@/lib/validation-schema";
+import { responseWrapper } from "@/utils/api-response-wrapper";
+import { NextRequest } from "next/server";
+
+export async function POST(req: NextRequest) {
+  try {
+    const { name, email, password } = await req.json();
+
+    if (name === null || name === "") {
+      return responseWrapper(400, null, "Name is required.");
+    }
```

Comment:
if (!name?.trim()) {
  return responseWrapper(400, null, "Name is required.");
} else if (!email?.trim()) {
  return responseWrapper(400, null, "Email is required.");
} else if (!password?.trim()) {
  return responseWrapper(400, null, "Password is required.");
}